Use of uninitialised var causes error in 3.7.0

## Configuration  ##

contents of minimised `.scalafmt.conf` file:
```
version = "3.7.0"
align.preset = most
runner.dialect = scala213source3
```

## Command-line parameters (required) ##

When I run scalafmt via CLI like this: `scalafmt` (or via mill or sbt)

## Steps

Given code like this:
```scala
object ScalaFormatRecreator extends App {
  private var s: String = _
}

```

## Problem

Here is the minimised recreator
```scala
org.scalafmt.cli.FailedToFormat: <path>/scalafmt-issue-recreator/src/main/scala/ScalaFormatRecreator.scala
Caused by: java.lang.NullPointerException: Cannot invoke "scala.meta.Tree.tokens(scala.meta.Dialect)" because "tree" is null
	at org.scalafmt.internal.FormatTokens.getHeadOpt(FormatTokens.scala:194)
	at org.scalafmt.internal.FormatTokens.tokenJustBeforeOpt(FormatTokens.scala:246)
	at org.scalafmt.internal.FormatWriter$FormatLocations.getAlignContainerParent(FormatWriter.scala:1293)
	at org.scalafmt.internal.FormatWriter$FormatLocations.getAlignContainer(FormatWriter.scala:1320)
	at org.scalafmt.internal.FormatWriter$FormatLocations.$anonfun$getAlignContainer$1(FormatWriter.scala:1340)
	at scala.Option.flatMap(Option.scala:283)
	at org.scalafmt.internal.FormatWriter$FormatLocations.getAlignContainer(FormatWriter.scala:1336)
	at org.scalafmt.internal.FormatWriter$FormatLocations.processLine$1(FormatWriter.scala:1165)
	at org.scalafmt.internal.FormatWriter$FormatLocations.alignmentTokens(FormatWriter.scala:1193)
	at org.scalafmt.internal.FormatWriter$FormatLocations.<init>(FormatWriter.scala:424)
	at org.scalafmt.internal.FormatWriter.getFormatLocations(FormatWriter.scala:139)
	at org.scalafmt.internal.FormatWriter.mkString(FormatWriter.scala:32)
	at org.scalafmt.Scalafmt$.$anonfun$doFormatOne$7(Scalafmt.scala:157)
	at scala.util.Success.flatMap(Try.scala:258)
	at org.scalafmt.Scalafmt$.$anonfun$doFormatOne$4(Scalafmt.scala:156)
	at scala.meta.parsers.Parsed.fold(Errors.scala:12)
	at scala.meta.parsers.Parsed.fold$(Errors.scala:11)
	at scala.meta.parsers.Parsed$Success.fold(Errors.scala:26)
	at org.scalafmt.Scalafmt$.org$scalafmt$Scalafmt$$doFormatOne(Scalafmt.scala:152)
	at org.scalafmt.Scalafmt$.doFormat(Scalafmt.scala:126)
	at org.scalafmt.Scalafmt$.formatCodeWithStyle(Scalafmt.scala:91)
	at org.scalafmt.Scalafmt$.$anonfun$formatCode$3(Scalafmt.scala:78)
	at scala.util.Success.fold(Try.scala:281)
	at org.scalafmt.Scalafmt$.formatCode(Scalafmt.scala:78)
	at org.scalafmt.cli.ScalafmtCoreRunner$.unsafeHandleFile(ScalafmtCoreRunner.scala:69)
	at org.scalafmt.cli.ScalafmtCoreRunner$.handleFile(ScalafmtCoreRunner.scala:53)
	at org.scalafmt.cli.ScalafmtCoreRunner$.$anonfun$run$5(ScalafmtCoreRunner.scala:36)
	at org.scalafmt.cli.ScalafmtCoreRunner$.$anonfun$run$5$adapted(ScalafmtCoreRunner.scala:35)
	at scala.collection.IterableOnceOps.foreach(IterableOnce.scala:575)
	at scala.collection.IterableOnceOps.foreach$(IterableOnce.scala:573)
	at scala.collection.AbstractIterator.foreach(Iterator.scala:1300)
	at scala.collection.parallel.ParIterableLike$Foreach.leaf(ParIterableLike.scala:938)
	at scala.collection.parallel.Task.$anonfun$tryLeaf$1(Tasks.scala:52)
	at scala.runtime.java8.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.scala:18)
	at scala.util.control.Breaks$$anon$1.catchBreak(Breaks.scala:97)
	at scala.collection.parallel.Task.tryLeaf(Tasks.scala:55)
	at scala.collection.parallel.Task.tryLeaf$(Tasks.scala:49)
	at scala.collection.parallel.ParIterableLike$Foreach.tryLeaf(ParIterableLike.scala:935)
	at scala.collection.parallel.AdaptiveWorkStealingTasks$AWSTWrappedTask.compute(Tasks.scala:152)
	at scala.collection.parallel.AdaptiveWorkStealingTasks$AWSTWrappedTask.compute$(Tasks.scala:148)
	at scala.collection.parallel.AdaptiveWorkStealingForkJoinTasks$AWSFJTWrappedTask.compute(Tasks.scala:304)
	at java.base/java.util.concurrent.RecursiveAction.exec(RecursiveAction.java:194)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:373)
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1182)
	at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1655)
	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1622)

```

## Expectation

It should not throw an exception

## Workaround

Initialising the var where it is created fixes the issue. However this is not always desirable.
Also removing the `align.preset = most` allows the scalafmt to complete (although obviously not format as desired)

## Notes

Use of `var` is obviously discouraged most of the time, but in Jmh benchmarks or highly optimised code it has a place, so should be supported

--- a/project/Dependencies.scala
+++ b/project/Dependencies.scala
@@ -6,7 +6,7 @@ import org.portablescala.sbtplatformdeps.PlatformDepsPlugin.autoImport._
 
 object Dependencies {
   val metaconfigV = "0.11.1"
-  val scalametaV  = "4.7.2"
+  val scalametaV  = "4.7.3"
   val scalacheckV = "1.17.0"
   val coursier    = "1.0.3"
   val munitV      = "0.7.29"

--- a/scalafmt-tests/src/test/resources/align/DefaultWithAlign.stat
+++ b/scalafmt-tests/src/test/resources/align/DefaultWithAlign.stat
@@ -2029,3 +2029,9 @@ object a {
       eeeee: Type6
   ): Future[Option[ReturnType]] = { ??? }
 }
+<<< #3439
+object ScalaFormatRecreator extends App {
+  private var s: String = _
+}
+>>>
+org.scalafmt.util.FormatException: java.lang.NullPointerException: Cannot invoke "scala.meta.Tree.tokens(scala.meta.Dialect)" because "tree" is null

--- a/scalafmt-tests/src/test/resources/align/DefaultWithAlign.stat
+++ b/scalafmt-tests/src/test/resources/align/DefaultWithAlign.stat
@@ -2034,4 +2034,6 @@ object ScalaFormatRecreator extends App {
   private var s: String = _
 }
 >>>
-org.scalafmt.util.FormatException: java.lang.NullPointerException: Cannot invoke "scala.meta.Tree.tokens(scala.meta.Dialect)" because "tree" is null
+object ScalaFormatRecreator extends App {
+  private var s: String = _
+}

